src/core/LegalDocument.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ILegalDocument,
+  LegalDocumentCategory,
+  LegalFile,
+  LegalFileType,
+  MultiResourceCreationResponse,
+  PostLegalDocumentDTO,
+  PostLegalFileDTO,
+} from '../lib';
+import { RequestBuilder } from '../RequestBuilder';
+import { Document } from './Document';
+
+/**
+ * Legal document instance
+ * Cf: https://developers.algoan.com/api/#operation/getLegalDocuments
+ */
+export class LegalDocument extends Document implements ILegalDocument {
+  /**
+   * Category of the legal document
+   */
+  public category: LegalDocumentCategory;
+
+  /**
+   * If not defined, it takes all values for a given category by default.
+   */
+  public validFileTypes: LegalFileType[];
+
+  public files: LegalFile[];
+
+  constructor(params: ILegalDocument, folderId: string, private readonly requestBuilder: RequestBuilder) {
+    super(params, folderId);
+    this.category = params.category;
+    this.validFileTypes = params.validFileTypes;
+    this.files = params.files;
+  }
+
+  /**
+   * Push a new legal document in the folder's list of documents.
+   *
+   * @param requestBuilder Service account request builder
+   * @param folderId the id of the folder in which documents are created
+   * @param body the legal documents to add
+   */
+  public static async createLegalDocuments(
```

Comment:
The `LegalDocuments` is not required, since it will be called with the LegalDocument class

```suggestion
  public static async create(
```

Reply:
Right!